CenterForOpenScience/osf.io
# [feature] Files Page Optimizations [OSF-8677]

#### Purpose
- The files page is suuuuuper slow to load for large projects. This PR improves performance (slightly) by reducing the number of queries being made and not loading all children beyond first level unless they are manually expanded.
- Note: This PR only makes a *slight* improvement because we're still automatically expanding top-level children on page load. For an even faster page load, we should NOT automatically expand children, but that's a larger UI/UX decision that needs to be discussed amongst Product. 

#### Changes
- Removes the osf-logo spinner in favor of ball-scale. 
- Rewrote a bit of rubeus.py to make fewer queries (uses subqueries, annotations, other fun stuff)
  - Additionally, only children of the parent project and first-level children are serialized in the files grid response. 
- Adds functionality to the treebeard grid to only request data about second level children when manually expanded by the user.

#### Side Effects
- None expected.

#### QA Notes 
- The files page UI should appear exactly the same when the page loads (i.e. addons and first-level children are automatically expanded). When expanding a second-level (or greater) child, a request is made to retrieve the children and you'll see the loading icon until the component expands. 
- This PR *should* only affect the way the file tree loads and how folders/addons expand and collapse, although a basic sweep of functionality (moving files, creating folders, etc) wouldn't be a bad idea. Pay special attention to permissions (i.e. that component names aren't shown to non-contributors). Also, please check different node types (forks, links, registrations) in the file tree. 

#### Stats 🐱 
```
/api/v1/project/<node>/files/grid/

Project with 42 top-level children, 93 totol children
    BEFORE
	num queries: 4131
	time (avg): 64.2 s
    AFTER 
	num queries: 1148
	time: 25.03 s	

Project with 1 top-level child 
    BEFORE 
	num queries: 47
	time: 5.1 s
    AFTER 
	num queries: 41
	time: 4.76 s
```

#### Ticket
- [OSF-8677](https://openscience.atlassian.net/browse/OSF-8677)
